flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointMetricsBuilder.java

```diff
@@ -69,8 +86,9 @@ public long getCheckpointStartDelayNanos() {
 	}
 
 	public CheckpointMetrics build() {
+		checkState(alignmentDurationNanos.isDone());
 		return new CheckpointMetrics(
-			alignmentDurationNanos,
+			FutureUtils.getWithoutException(alignmentDurationNanos),
```

Comment:
That's a good idea. We could move this check to some `FutureUtils#isCompletedNormally`, it would also be good to chain the original exception to the checkstate there.